src/core/job.c

```diff
@@ -839,6 +858,21 @@ static void job_fail_dependencies(Unit *u, UnitDependency d) {
         }
 }
 
+static int job_save_pending_finished_job(Job *j) {
+        Manager *m = j->manager;
+
+        if (!m->pending_finished_jobs) {
+                int r;
+
+                r = set_ensure_allocated(&m->pending_finished_jobs, NULL);
```

Comment:
`set_ensure_allocated` already does the check above and does nothing if the set was allocated before.